mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -24,6 +25,21 @@ def SYCL_Dialect : Dialect {
   let name = "sycl";
   let cppNamespace = "::mlir::sycl";
   let useDefaultTypePrinterParser = 1;
+  let extraClassDeclaration = [{
+    SYCLContext syclCtx;
```

Comment:
Can this be the new `MethodRegistry` rather than a `SYCLContext` ?

Reply:
I thought it was a good idea to have a `SYCLContext` holding relevant data structures the same way dialects have `MLIRContext` for that.